### PR TITLE
Add Merge Sort

### PR DESCRIPTION
**Pull request by gr-akshaya**

### CODE REVIEW

**Comment by Matrix278** [comment]
Wow, nice, but can you rename the folder to be as mergeSort not MergeSort

File: Mergesort/mergesort.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+RightArray[j] = arr[mid + 1+ j];  
+  
+  
+int i = 0, j = 0;  
+int k = beg;  
+while (i<l&&j<r)  
+{  
+if (LeftArray[i] <= RightArray[j])  
+{  
+arr[k] = LeftArray[i];
```
Comment (Matrix278) [inline Mergesort/mergesort.java]: And can you rewrite this java file, to have tabs and all was looking readable

Comment (gr-akshaya) [reply Mergesort/mergesort.java]: Sure

Comment (gr-akshaya) [reply Mergesort/mergesort.java]: Renamed and Added Tabs

**Comment by Matrix278** [comment]
Is the folder renamed? 🤔 Can't see this change

**Comment by Matrix278** [comment]
Anyway I think thats ok, I can change that if its not renamed

**Comment by Matrix278** [comment]
Ah no, java code is not working, @akshayacode please check that
```
$javac MyMergeSort.java
MyMergeSort.java:9: error: not a statement
       intLeftArray[] = new int [l];  
                   ^
MyMergeSort.java:9: error: ';' expected
       intLeftArray[] = new int [l];  
                     ^
MyMergeSort.java:9: error: ']' expected
       intLeftArray[] = new int [l];  
                                 ^
MyMergeSort.java:9: error: ';' expected
       intLeftArray[] = new int [l];  
                                  ^
MyMergeSort.java:10: error: not a statement
       intRightArray[] = new int [r];  
                    ^
MyMergeSort.java:10: error: ';' expected
       intRightArray[] = new int [r];  
                      ^
MyMergeSort.java:10: error: ']' expected
       intRightArray[] = new int [r];  
                                  ^
MyMergeSort.java:10: error: ';' expected
       intRightArray[] = new int [r];  
                                   ^
MyMergeSort.java:63: error: not a statement
        intarr[] = {6,5,3,1,8,7,2,4};  
              ^
MyMergeSort.java:63: error: ';' expected
        intarr[] = {6,5,3,1,8,7,2,4};  
                ^
MyMergeSort.java:63: error: not a statement
        intarr[] = {6,5,3,1,8,7,2,4};  
                    ^
MyMergeSort.java:63: error: ';' expected
        intarr[] = {6,5,3,1,8,7,2,4};  
                     ^
12 errors
```

**Comment by gr-akshaya** [comment]
I have resolved errors can u plz  check that

**Comment by Matrix278** [comment]
Thank you for contribution

**Pull request merged**